tensorflow_probability/python/distributions/relaxed_onehot_categorical.py

```diff
@@ -407,15 +407,26 @@ def __init__(
         undefined statistics will return NaN for this statistic.
       name: A name for this distribution (optional).
     """
+
+  
+
     dist = ExpRelaxedOneHotCategorical(temperature,
                                        logits=logits,
                                        probs=probs,
                                        validate_args=validate_args,
                                        allow_nan_stats=allow_nan_stats)
+
+    self._temperature = dist.temperature
```

Comment:
Why do you need to this? Doesn't `self.distribution.temperature` not work?

Reply:
You're right, I got a bit confused. Changing now.